src/components/subscriptions/edit/EditSubscription.js

```diff
@@ -0,0 +1,228 @@
+    const [mutateSubscriptionError, setMutateSubscriptionError] =
+        useState(null);
+    const { t } = useTranslation("subscriptions");
+
+    const preProcessPlanTypes = React.useCallback(
+        (data) => {
+            if (data?.result?.length > 0) {
+                setPlanTypes(data.result.map((type) => type.name));
+            }
+        },
+        [setPlanTypes]
+    );
+
+    const { mutate: mutateSubscription } = useMutation(
+        () => postSubscription(subscriptionSubmission),
+        {
+            onSuccess: (data) => {
+                const dataResult = data?.result[0];
+                const isFailed = dataResult?.failure_reason;
+                isFailed ? setFailureReason(isFailed) : setFailureReason(null);
+                setMutateSubscriptionError(isFailed ? dataResult : null);
+                if (!isFailed) {
+                    announce({
+                        text: "Success",
+                    });
+                    onClose();
+                }
+            },
+            onError: (err) => {
+                setMutateSubscriptionError(err);
+                console.log(err);
+            },
+        }
+    );
+
+    const handleSubmit = (values) => {
+        setSubscriptionSubmission(formatSubscriptions(values));
+        mutateSubscription();
+    };
+
+    const onCloseForm = () => {
+        onClose();
+        resetState();
+    };
+
+    const resetState = () => {
+        setMutateSubscriptionError(null);
+        setFailureReason(null);
+        setSubscriptionSubmission(null);
+    };
+
+    useQuery({
+        queryKey: [PLAN_TYPES_QUERY_KEY],
+        queryFn: () => getPlanTypes(),
+        enabled: true,
+        staleTime: Infinity,
+        cacheTime: Infinity,
+        onSuccess: preProcessPlanTypes,
+    });
+
+    return (
+        <>
+            <Formik
+                initialValues={mapPropsToValues(subscription)}
+                onSubmit={handleSubmit}
+                enableReinitialize={true}
+            >
+                {({ handleSubmit, ...props }) => {
+                    return (
+                        <Form>
+                            <DEDialog
+                                {...props}
+                                id={parentId}
+                                open={open}
+                                onClose={() => {
+                                    onCloseForm();
+                                    props.resetForm();
+                                }}
+                                fullWidth={true}
+                                title={
+                                    subscription
+                                        ? t("editSubscription")
+                                        : t("addSubscription")
+                                }
+                                actions={
+                                    <>
+                                        <Button
+                                            {...props}
```

Comment:
Would it be better to destructure resetForm out of props?

Reply:
You could, but it would be destructured from the props of the anonymous child component of this `Formik` tag.

Reusing my example from above, we can do it like this:
```jsx
            <Formik
                initialValues={mapPropsToValues(subscription)}
                onSubmit={handleSubmit}
                enableReinitialize={true}
            >
                {({ handleSubmit, resetForm }) => {
                    const onClose = () => {
                        onCloseForm();
                        resetForm();
                    };

                    return (
                        <Form>
                            <DEDialog
                                onClose={onClose}
```
If the `onClose` callback is defined like this, then it could also be reused in the `onClick` callback for this `Button` as `onClick={onClose}`.

I'm not seeing these `props` used anywhere else, so `resetForm` can go ahead and replace `...props`.